core/server/worker/src/test/java/alluxio/worker/block/CacheRequestManagerTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import alluxio.Sessions;
+import alluxio.client.file.FileSystemContext;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.grpc.CacheRequest;
+import alluxio.proto.dataserver.Protocol;
+import alluxio.util.network.NetworkAddressUtils;
+import alluxio.worker.grpc.GrpcExecutors;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Random;
+
+/**
+ * Unit tests for {@link CacheRequestManager}.
+ */
+public class CacheRequestManagerTest {
+  private CacheRequestManager mCacheRequestManager;
+  private Random mRandom;
+  private BlockWorker mBlockWorker;
+  private FileSystemContext mFsContext;
+
+  /**
+   * Sets up all dependencies before a test runs.
+   */
+  @Before
+  public void before() throws IOException {
+    mRandom = new Random();
+    mBlockWorker = mock(DefaultBlockWorker.class);
+    mFsContext = mock(FileSystemContext.class);
```

Comment:
in general, I am not a big fan of using mock to test important features.
Can you check `BlockWorkerDataReaderTest` and see if we can do similar setup to test Cache api?

Reply:
After I read BlockWorkerDataReaderTest and discussed with David, it seems like all the operation can be finished within alluxio space(within one blockWorker), but our case need UFS or remote worker which are not obvious to me how to implement in the unit test. One thing I can think of right now is testing the scenario that Block already cached in the worker.